airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/xmin/PostgresXminHandler.java

```diff
@@ -0,0 +1,195 @@
+          .get(fullyQualifiedTableName);
+      final List<String> selectedDatabaseFields = table.getFields()
+          .stream()
+          .map(CommonField::getName)
+          .filter(CatalogHelpers.getTopLevelFieldNames(airbyteStream)::contains)
+          .collect(Collectors.toList());
+
+      final AutoCloseableIterator<JsonNode> queryStream = queryTableXmin(selectedDatabaseFields, table.getNameSpace(), table.getName());
+      final AutoCloseableIterator<AirbyteMessage> recordIterator =
+          getRecordIterator(queryStream, streamName, namespace, emittedAt.toEpochMilli());
+      final AutoCloseableIterator<AirbyteMessage> recordAndMessageIterator = augmentWithState(recordIterator, pair);
+
+      iteratorList.add(augmentWithLogs(recordAndMessageIterator, pair, streamName));
+    }
+
+    return iteratorList;
+  }
+
+  private AutoCloseableIterator<JsonNode> queryTableXmin(
+                                                         final List<String> columnNames,
+                                                         final String schemaName,
+                                                         final String tableName) {
+    LOGGER.info("Queueing query for table: {}", tableName);
+    final AirbyteStreamNameNamespacePair airbyteStream =
+        AirbyteStreamUtils.convertFromNameAndNamespace(tableName, schemaName);
+    return AutoCloseableIterators.lazyIterator(() -> {
+      try {
+        final Stream<JsonNode> stream = database.unsafeQuery(
+            connection -> createXminQueryStatement(connection, columnNames, schemaName, tableName, airbyteStream),
+            sourceOperations::rowToJson);
+        return AutoCloseableIterators.fromStream(stream, airbyteStream);
+      } catch (final SQLException e) {
+        throw new RuntimeException(e);
+      }
+    }, airbyteStream);
+  }
+
+  private PreparedStatement createXminQueryStatement(
+      final Connection connection,
+      final List<String> columnNames,
+      final String schemaName,
+      final String tableName,
+      final AirbyteStreamNameNamespacePair airbyteStream) {
+    try {
+      LOGGER.info("Preparing query for table: {}", tableName);
+      final String fullTableName = getFullyQualifiedTableNameWithQuoting(schemaName, tableName,
+          quoteString);
+
+      final String wrappedColumnNames = RelationalDbQueryUtils.enquoteIdentifierList(columnNames, quoteString);
+      final String sql = String.format("SELECT %s FROM %s WHERE xmin::text::bigint > ?",
+          wrappedColumnNames, fullTableName);
+
+      final PreparedStatement preparedStatement = connection.prepareStatement(sql.toString());
+
+      final XminStatus currentStreamXminStatus = xminStateManager.getXminStatus(airbyteStream);
+      if (currentStreamXminStatus != null) {
+        preparedStatement.setLong(1, currentStreamXminStatus.getXminXidValue());
+      } else {
+        preparedStatement.setLong(1, 0L);
```

Comment:
This would effectively be a full refresh sync? is this expected to ever happen?
Maybe log a warning here?

Reply:
If a user adds a new stream, there won't be any xmin state associated with that stream and we'll have to issue a full refresh in that case (for that particular table)

Reply:
So in this case it is full refresh by xmin.
Does it differ from "regular" full refresh?
Does it matter that we go into a different flow?

Reply:
Logically, it is the same flow as the vanilla full refresh flow. It's just contained here when we do also want to query `xmin` status with every query for intermediate state checkpointing in future PRs